File: Models/CapitalAllowance.cs
using System;
namespace TaxComputationAPI.Models
{
    public class CapitalAllowance
    {

        public int Id {get;set;}
        public string TaxYear { get; set; }
        public decimal  OpeningResidue { get; set; }
        public decimal Addition { get; set; }
        public decimal Disposal { get; set; }
        public decimal Initial { get; set; }
        public decimal Annual { get; set; }
        public decimal Total { get;set;}
        public decimal ClosingResidue { get; set; }
        public int YearsToGo { get; set; }
        public int CompanyId { get; set; }
        public int AssetId { get; set; }
        public string YearId {get;set;}

        public int NumberOfYearsAvailable {get;set;}
        public string CompanyCode {get;set;}
        
        public  string Year { get; set; }

        public string  Channel {get;set;} 

    }
}

File: taxcomputation-spa/src/apis/CapitalAllowance.js
import axios from "axios";

export const capitalAllowanceAdding = async ({
  companyId,
  taxYear,
  assetId,
  openingResidue,
  addition,
  disposal,
  initial,
  annual,
  total,
  closingResidue,
  yearsToGo
}) => {
  try {
    return await axios.post("/api/CapitalAllowance", {
      companyId,
      taxYear,
      assetId,
      openingResidue,
      addition,
      disposal,
      initial,
      annual,
      total,
      closingResidue,
      yearsToGo
    });
  } catch (error) {
    throw error;
  }
};

export const capitalAllowanceViewData = async ({ companyId, assetId }) => {
  try {
    const { data } = await axios.get(`/api/CapitalAllowance/${companyId}/${assetId}`);
    return data;
  } catch (error) {
    throw error;
  }
};

export const capitalAllowanceSummaryData = async ({ companyId }) => {
  try {
    const { data } = await axios.get(
      `/api/CapitalAllowanceSummary/companyId?companyId=${companyId}`
    );
    return data;
  } catch (error) {
    throw error;
  }
};

export const capitalAllowanceDelete = async ({ id }) => {
  try {
    const { data } = await axios.delete(`/api/CapitalAllowance/${id}`);
    return data;
  } catch (error) {
    throw error;
  }
};
